src/hip_memory.cpp

```diff
@@ -1542,44 +1542,73 @@ void ihipMemsetKernel(hipStream_t stream, T* ptr, T val, size_t sizeBytes) {
                        sizeBytes, std::move(val));
 }
 
-
-// TODO-sync: function is async unless target is pinned host memory - then these are fully sync.
-hipError_t hipMemsetAsync(void* dst, int value, size_t sizeBytes, hipStream_t stream) {
-    HIP_INIT_SPECIAL_API((TRACE_MCMD), dst, value, sizeBytes, stream);
-
+typedef enum ihipMemsetCopyDataType {
+    ihipMemsetCopyDataTypeChar   = 0,
+    ihipMemsetCopyDataTypeShort  = 1,
+    ihipMemsetCopyDataTypeInt    = 2
+}ihipMemsetCopyDataType;
+
+hipError_t ihipMemset(void* dst, int  value, size_t sizeBytes, hipStream_t stream, enum ihipMemsetCopyDataType copyDataType  )
```

Comment:
I don't see any calls to ihipMemset with the copyDataType being set to Short or Int in the rest of the code. Are those changes coming in later?

Reply:
Yes, those changes will be coming today with support of hipMemsetD16/32 variants.